sqlitecluster/SQLiteNode.h

```diff
@@ -205,10 +213,32 @@ class SQLiteNode : public STCPNode {
     chrono::steady_clock::time_point _lastNetStatTime;
 
     // Handler for transaction messages.
-    void handleBeginTransaction(Peer* peer, const SData& message);
-    void handleCommitTransaction(Peer* peer, const SData& message);
-    void handleRollbackTransaction(Peer* peer, const SData& message);
+    void handleBeginTransaction(SQLite& db, Peer* peer, const SData& message);
+    void handleCommitTransaction(SQLite& db, Peer* peer, const uint64_t commandCommitCount, const string& commandCommitHash);
+    void handleRollbackTransaction(SQLite& db, Peer* peer, const SData& message);
 
     WallClockTimer _syncTimer;
-    uint64_t _handledCommitCount;
+    atomic<uint64_t> _handledCommitCount;
+
+    // List of threads that are handling replication. These are transient and only exist while FOLLOWING.
+    list<thread> _replicationThreads;
+
+    // State variable that indicates when the above threads should quit.
+    atomic<bool> _replicationThreadsShouldExit;
+
+    // Mutex that guards the map of hashes for transactions currently being replicated.
+    mutex _replicationHashMutex;
+    map<uint64_t, string> _replicationHashes;
+
+    // Mutex, condition variable, and queue object for command 
+    mutex _replicationCommandMutex;
+    condition_variable _replicationCV;
```

Comment:
I'm surprised `SSynchronizeQueue` doesn't have a `condition_variable` built in; it feels weird that this needs to be handled externally.  Same for `_replicationCommandMutex` -- again, it feels like this is a common pattern we'd need for every `SSynchronizedQueue`.  (And even if it's *not* used everywhere, it's cheap to include and ignore.)

Reply:
This might be better, just add `wait` and `notify` to the queue. It'd certainly be cleaner here in this code, and encapsulate that functionality better. I'll add it there and remove it here.